web/server/codechecker_server/api/report_server.py

```diff
@@ -1901,9 +1901,15 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
 
                 # Get report details if it is required.
                 report_details = {}
-                if get_details:
-                    report_ids = [r[0].id for r in query_result]
+                blame_infos = {}
+                if get_details and len(query_result):
+                    report_ids, blames = zip(*[
+                        (
+                            r[0].id,
+                            (r[0].id, self.getBlameInfo(r[0].file_id))
+                        ) for r in query_result])
                     report_details = get_report_details(session, report_ids)
+                    blame_infos = dict(blames)
```

Comment:
`getRunResults` is already stomach-churningly long. Can we put these lines a new function instead?